packages/svelte-cloudinary/src/components/CldVideoPlayer.svelte

```diff
@@ -1,214 +1,191 @@
+	import { loadScript } from '../helpers/scripts';
+	import { mergeGlobalConfig } from '../config';
+	import { onDestroy, onMount } from 'svelte';
 
-	function getPlayerRefs() {
-		return {
-			player: playerRef,
-			video: videoRef
-		};
-	}
+	const PLAYER_VERSION = '1.11.1';
 
-	//@ts-ignore
-	function onLoadError(e) {
-		console.error(`Failed to load Cloudinary Video Player: ${e.message}`);
+	type $$Props = CldVideoPlayerProps;
+	$: ({ config, id, ...videoPlayerOptions } = $$props as CldVideoPlayerProps);
+
+	$: options = getVideoPlayerOptions(
+		videoPlayerOptions,
+		mergeGlobalConfig(config).config,
+	);
+
+	let loaded =
+		typeof window != 'undefined' && !!window.cloudinary?.videoPlayer;
+
+	export let videoElement: HTMLVideoElement | undefined = undefined;
+	export let player: CloudinaryVideoPlayer | undefined = undefined;
+
+	$: if (videoElement && loaded && !player) {
+		player = window.cloudinary?.videoPlayer?.(videoElement, options);
+
+		player?.on('error', () =>
+			videoPlayerOptions.onError?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('loadeddata', () =>
+			videoPlayerOptions.onDataLoad?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('loadedmetadata', () =>
+			videoPlayerOptions.onMetadataLoad?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('pause', () =>
+			videoPlayerOptions.onPause?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('play', () =>
+			videoPlayerOptions.onPlay?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
+
+		player?.on('ended', () =>
+			videoPlayerOptions.onEnded?.({
+				player: player!,
+				video: videoElement!,
+			}),
+		);
 	}
 
 	onMount(() => {
-		if (!window.cloudinary?.videoPlayer) {
-			return loadCloudinary({ type: 'video', onLoad: handleOnLoad, onError: onLoadError });
-		}
-		return handleOnLoad();
+		loadScript({
+			src: `https://unpkg.com/cloudinary-video-player@${PLAYER_VERSION}/dist/cld-video-player.min.js`,
+			onLoad() {
+				loaded = true;
+			},
+			onError() {
+				console.error('Failed to load the Cloudinary Video Player SDK');
+			},
+		});
+	});
+
+	onDestroy(() => {
+		player?.dispose();
 	});
 </script>
 
 <svelte:head>
 	<link
-		href={`https://unpkg.com/cloudinary-video-player@${version}/dist/cld-video-player.min.css`}
-		rel="stylesheet"
-	/>
+		href="https://unpkg.com/cloudinary-video-player@{PLAYER_VERSION}/dist/cld-video-player.min.css"
+		rel="stylesheet" />
 </svelte:head>
 
-<div style="width:100%;aspect-ratio:{$$props.width} / {$$props.height}">
-	<video bind:this={videoRef} id={playerId} class={playerClassName} {width} {height}>
+<div
+	style="width: 100%;"
+	style:aspect-ratio="{options.width} / {options.height}">
+	<video
+		id={id || options.publicId}
+		width={options.width}
+		height={options.height}
+		bind:this={videoElement}
+		class="cld-video-player cld-fluid">
 		<track kind="captions" />
```

Comment:
looks like this was here before. i never added this to mine 🤔  just pointing it out. will have to look into it

Reply:
another thing left over from v1, can remove if it's not necessary

Reply:
i suspect it's not necessary. im not seeing it with a quick search in the docs under subtitles. maybe remove and make a note on it?